types/mempool/nonce.go

```diff
@@ -0,0 +1,110 @@
+package mempool
+
+import (
+	"fmt"
+
+	huandu "github.com/huandu/skiplist"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+// nonceMempool is a mempool that keeps transactions sorted by nonce.  Transactions with the lowest nonce globally
+// are prioritized.  Transactions with the same nonce are prioritized by sender address. Fee/gas based
+// prioritization is not supported.
+type nonceMempool struct {
+	txQueue *huandu.SkipList
```

Comment:
Just want to point out that using a skip list this for this type of implementation might be overkill. You could just use a sorted `[]Tx` array, or a wrapper around `Tx` so you don't have to do anything extra to get the sender and nonce, etc..

Reply:
>You could just use a sorted []Tx array

I think this is an efficient way to maintain a sorted list.  Using an array we'd still need to sort in insert, what are you proposing for maintaining the sort?